src/main/java/io/kestra/storage/s3/S3Storage.java

```diff
@@ -117,6 +117,23 @@ public List<FileAttributes> list(String tenantId, URI uri) throws IOException {
         }
     }
 
+    private Stream<String> keysForPrefix(String prefix, boolean recursive) {
+        ListObjectsV2Request request = ListObjectsV2Request.builder()
+            .bucket(s3Config.getBucket())
+            .prefix(prefix)
+            .build();
+        List<S3Object> contents = s3Client.listObjectsV2(request).contents();
+        return contents.stream()
+            .map(S3Object::key)
+            .filter(key -> {
+                key = key.substring(prefix.length());
+                // Remove recursive result and requested dir
+                return !key.isEmpty()
+                    && !Objects.equals(key, prefix)
+                    && (recursive || Path.of(key).getParent() == null);
```

Comment:
I don't understand why you filter on the recursive boolean here, if this boolean is false any call to the method will filter everything.

`new File()` could be replaced by `new Path()` I think.